src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.ConvertToInteger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Intrinsics;
+
+namespace System.Numerics.Tensors
+{
+    public static partial class TensorPrimitives
+    {
+        /// <summary>
+        /// Copies <paramref name="source"/> to <paramref name="destination"/>, converting each <typeparamref name="TFrom"/>
+        /// value to a <typeparamref name="TTo"/> integer type using saturation on overflow.
+        /// </summary>
+        /// <param name="source">The source span from which to copy values.</param>
+        /// <param name="destination">The destination span into which the converted values should be written.</param>
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <remarks>
+        /// <para>
+        /// This method effectively computes <c><paramref name="destination" />[i] = TTo.ConvertToInteger(<paramref name="source"/>[i])</c>.
+        /// </para>
+        /// </remarks>
+        public static void ConvertToInteger<TFrom, TTo>(ReadOnlySpan<TFrom> source, Span<TTo> destination)
+            where TFrom : IFloatingPoint<TFrom>
+            where TTo : IBinaryInteger<TTo> =>
+            InvokeSpanIntoSpan<TFrom, TTo, ConvertToIntegerOperator<TFrom, TTo>>(source, destination);
+
+        /// <summary>TFrom.ConvertToInteger&lt;TTo&gt;(x)</summary>
+        internal readonly struct ConvertToIntegerOperator<TFrom, TTo> : IUnaryOperator<TFrom, TTo>
+            where TFrom : IFloatingPoint<TFrom>
+            where TTo : IBinaryInteger<TTo>
+        {
+            public static TTo Invoke(TFrom x) =>
+#if NET9_0_OR_GREATER
+                TFrom.ConvertToInteger<TTo>(x);
+#else
+                TTo.CreateSaturating(x);
+#endif
```

Comment:
Is this desirable? Or should ConvertToInteger{Native} only be exposed on .NET 9+?

Reply:
> it's a bit confusing if we have an API like this on TensorPrimitives but don't expose corresponding support on the scalar types.

Should we do that for others where the method isn't available on the primitive downlevel? There are a bunch.

Reply:
> There are a bunch

I take that back. It's that there aren't the corresponding Vector methods for a bunch